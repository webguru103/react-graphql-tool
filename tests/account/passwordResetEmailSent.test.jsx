/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import 'jest-dom/extend-expect';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { withContext } from '../test-helpers/contextWrapper';
import PasswordResetEmailSent from '../../src/account/forgotPassword/passwordResetEmailSent';

describe('Password Reset Email Sent', () => {
  afterEach(cleanup);
  const Component = withContext(PasswordResetEmailSent);
  test('renders', () => {
    const { container } = render(<Component email="test@dealtap.ca" />);
    expect(container).toMatchSnapshot();
  });

  test('request is made on click resend button', () => {
    const handleResendEmail = jest.fn();
    const { getByTestId } = render(<Component
      email="test@dealtap.ca"
      sendResetPasswordEmail={handleResendEmail}
    />);
    const resendButton = getByTestId('submit-button');

    fireEvent(
      resendButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );

    expect(handleResendEmail).toHaveBeenCalledTimes(1);
  });
});
