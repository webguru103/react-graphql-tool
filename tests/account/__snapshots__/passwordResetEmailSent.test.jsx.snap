// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Reset Email Sent renders 1`] = `
<div>
  <div
    class="Apollo-SendResetPasswordEmailHOC--resetEmailSent-1"
  >
    <h3
      class="MuiTypography-root-4 MuiTypography-subheading-11"
    >
      A password reset email was sent to your email address. Please check your email.
    </h3>
    <div
      class="Apollo-SendResetPasswordEmailHOC--buttons-2"
    >
      <button
        class="MuiButtonBase-root-60 MuiButton-root-34 MuiButton-contained-45 MuiButton-containedPrimary-46 ButtonC-primary-31 MuiButton-raised-48 MuiButton-raisedPrimary-49 ButtonC-button-30 Apollo-SendResetPasswordEmailHOC--button-3"
        data-testid=""
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-35"
        >
          Login
        </span>
      </button>
      <button
        class="MuiButtonBase-root-60 MuiButton-root-34 MuiButton-contained-45 MuiButton-containedSecondary-47 ButtonC-secondary-32 MuiButton-raised-48 MuiButton-raisedSecondary-50 ButtonC-button-30 Apollo-SendResetPasswordEmailHOC--button-3"
        data-testid="submit-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-35"
        >
          Resend Email
        </span>
      </button>
    </div>
  </div>
</div>
`;
